packages/Input/src/Input.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useEffect } from 'react';
+import cx from 'classnames';
+
+import './input.css';
+
+const KeyCodes = {
+  Comma: 44,
+  Minus: 45,
+  Nine: 57,
+  Period: 46,
+  Plus: 43,
+  Zero: 48,
+};
+
+export type InputSize = 'small';
+
+export type InputType = 'email' | 'text' | 'password' | 'number';
+
+export interface InputProps extends React.ComponentPropsWithRef<'input'> {
+  // Add class names to the surrounding DOM container.
+  className?: string;
+  // Form.ValidatedField state. True if has error.
+  error?: boolean;
+  // Specifies the type to display
+  type?: InputType;
+  // Input size.
+  inputSize?: InputSize;
+  // Specifies the value inside the input.
+  value?: string | number;
+  // True if you need the input to be focus on page load.
+  autoFocus?: boolean;
+  // True if you need the input to be readonly.
+  readOnly?: boolean;
+  // Function called when the value changes.
+  onChange?: (e: any) => void;
+  // Add a data-test tag for automated tests
+  dataTest?: string;
+}
+
+const Input: React.FunctionComponent<InputProps> = React.forwardRef(
+  (props: InputProps, ref: React.Ref<any>) => {
+    const {
+      className,
+      error,
+      type = 'text',
+      inputSize,
+      value,
+      autoFocus,
+      readOnly,
+      onChange,
+      dataTest,
+      ...rest
+    } = props;
+
+    const classes = cx(
+      'ids-input',
+      className,
+      inputSize && `ids-input--${inputSize}`,
+      error,
```

Comment:
Il faut respecter la nomenclature des className soit préfixe `ids` et suive la méthodologie BEM `input--error`
```suggestion
      {
        'ids-input--error': error,
      }
```